### ISSUE TITLE
Automate check for core-> providers deps

### ISSUE THREAD

**Issue by potiuk** [apache/airflow#11435]
We should automate a check for dependencies between the core and providers. This should be CI-automated check that will prevent people from accidentally creating a dependency on any of the providers.

Linked issue #12247

**Comment by turbaszek** [comment]
There's a dependency on kubernetes in example dags:
https://github.com/apache/airflow/blob/ae397df3360a61d081579ac838293ae440c81cb3/airflow/example_dags/example_kubernetes_executor_config.py#L23

which is not required dependency for Airflow. If users don't have installed kubernetes package then they will face errors when using example dags as mentioned in #12247.

**Comment by eladkal** [comment]
I believe this one is completed as we are checking against min provider versions

**Comment by potiuk** [comment]
Not really. This one is to make sure that in "core" code and tests providers are not used. This is not yet entirely true because even if FAB is in a separate provider. This should be fixed by separating FAB out (cc: @jedcunningham) in AIP-79, because some tests and some core files are referring to FAB provider.

 I think we also have some K8S and celery provider references in core tests + some of the tests in core "always" are actually expecting providers to be installed and will fail if they are not.

This one is basically a prerequisite if we would want to move providers out in sources (we don't for now) - so no high priority, it's more something that eventually should happen.
